ndcube/extra_coords/table_coord.py

```diff
@@ -10,12 +10,71 @@
 from astropy.coordinates import SkyCoord
 from astropy.modeling import models
 from astropy.modeling.models import tabular_model
+from astropy.modeling.tabular import _Tabular
 from astropy.time import Time
 from astropy.wcs.wcsapi.wrappers.sliced_wcs import combine_slices, sanitize_slices
 
 __all__ = ['TimeTableCoordinate', 'SkyCoordTableCoordinate', 'QuantityTableCoordinate']
 
 
+class Length1Tabular(_Tabular):
+    _input_units_allow_dimensionless = True
+    _has_inverse_bounding_box = True
+    _separable = True
+
+    n_inputs = 1
+    n_outputs = 1
+
+    lookup_table = np.zeros([1])
+    points = np.zeros([1])
+
+    def __init__(self, points=None, lookup_table=None, pixel_width=None,
+                 method='linear', bounds_error=True, fill_value=np.nan, **kwargs):
+        if len(lookup_table) != 1:
+            raise ValueError("lookup_table must have length 1.")
+        super().__init__(points=points, lookup_table=lookup_table, method=method,
+                         bounds_error=bounds_error, fill_value=fill_value, **kwargs)
+        self._pixel_width = pixel_width  # Width of pixel in world units.
+
+    def evaluate(self, x):
+        output = np.full(x.shape, self.fill_value)
+        x_unit = self.input_units['x']
+        output[np.logical_and(x >= -0.5*x_unit, x < 0.5*x_unit)] = self.lookup_table[0].value
+        return output * self.lookup_table.unit
+
+    @property
+    def inverse(self):
+        return InverseLength1Tabular(points=self.points[0], lookup_table=self.lookup_table,
+                                     pixel_width=self._pixel_width, method=self.method,
+                                     bounds_error=self.bounds_error, fill_value=self.fill_value)
+
+
+class InverseLength1Tabular(Length1Tabular):
+    def __init__(self, **kwargs):
+        points = kwargs.pop("points", None)
+        lookup_table = kwargs.pop("lookup_table", None)
+        super().__init__(points=lookup_table, lookup_table=points, **kwargs)
+        if self._pixel_width is None:
+            self._pixel_width = 0 * lookup_table.unit
```

Comment:
Defaulting to 0 here? We aren't overriding this anywhere?

Reply:
It's given by the `pixel_width` kwarg, defined in the init of `Length1Tabular` which this class inherits.

Reply:
but I mean we aren't populating it in our code anywhere, so it's always going to use the default?

Reply:
Unless the user sets the `pixel_width` kwarg, yes.  But that's how default values work no?